launch_testing/test/launch_testing/new_hooks/test_function_scope.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import launch
+import launch_testing
+
+import pytest
+
+
+@pytest.fixture(scope='module')
+def order():
+    return []
+
+
+@launch_testing.pytest.fixture
+def launch_description():
+    return launch.LaunchDescription([
+        launch_testing.util.KeepAliveProc(),
+        launch_testing.actions.ReadyToTest(),
+    ])
+
+
+# TODO(ivanpauno)
+# We cannot get variables from the dictionary returned by launch_description
+# because we're removing the fixture before running the tests.
+# Maybe we can delete this feature, and use generators/asyncgens.
+# We can always get the variables returned by the launch_testing fixture there.
```

Comment:
@ivanpauno meta: this is no longer true, is it?

Reply:
exactly, comment deleted in https://github.com/ros2/launch/pull/528/commits/dda4cd9f3e48a7d30d44c8dae79d59dd63c864af.